### ISSUE TITLE
Runic altar requires brake and replace

### ISSUE THREAD

**Issue by Tacete** [FTBTeam/FTB-Modpack-Issues#6319]
### Modpack

FTB Interactions Remastered

### Modpack version

1.9.0

### Launcher

FTB App

### Has the pack been modified

Yes

### Describe the bug

The runic altar from botania will not accept items once the "right click with an empty hand to add back the last recipe" message has timed out.

### Steps to reproduce

craft something in the runic altar (I was making runes of sacrifice but another player on my server had the same issue with botania runes). 
Then wait until you no longer have the option to replace the craft.
attempt to place or drop items on the altar

### Expected behaviour

Altar accepts items

### Additional information

only modifications I ahve made are enabling maintenance and the clean room in the GTCEu configs

**Comment by ArtpokeBr** [comment]
weird, I don't see this problem in my instance

**Comment by Tacete** [comment]
I am on a server in the overworld if that makes any difference. The altar just sits and keep spawning the blue lightning particle effects and wont take any items
![image](https://github.com/user-attachments/assets/e6d3eb25-ac23-4579-a382-7d0c799ad36a)
![image](https://github.com/user-attachments/assets/a67ca6ea-53ec-46ce-b108-28178300c5dd)

**Comment by CaptainFord** [comment]
I have this problem #6528 too, but I noticed that it actually starts a new blank craft after the first one is completed, as the mana spreaders begin firing again and I also hear the sounds that indicates the start of a new craft overlaid with the sound of finishing the last one. Are you seeing that too?

**Comment by Tacete** [comment]
I am seeing this as well the mana spreaders shoot a few bursts and I hear
the sound for a bit.  It's like a new empty craft starts and just never
completes

**Comment by CaptainFord** [comment]
So weird. If it's happening to just us and not others, it could be something environmental. I know I'm running it with, let's see "OpenJDK 64-Bit Server VM" rather than an Oracle JRE. I wonder if something like that might be the cause.

**Comment by Tacete** [comment]
This continues to be strange because I am using the JRE.

**Comment by CaptainFord** [comment]
Looking at the way Botania's code is structured, the block's state seems to hinge simply on the field `manaToGet`. If it's zero, then no recipe is in progress, if it's >0 then it gathers mana and if `mana >= manaToGet` then it does the lightning and allows the recipe to be completed.

So all that needs to happen for this behavior to be exhibited is for `mana` to be reset (either to zero or `mana -= manaToGet`) and for `manaToGet` to not be reset to zero. 

And looking deeper, `manaToGet` is actually set every tick based on `currentRecipe`, which is not checked again in the update loop after it has been selected. So to see this behavior all that needs to happen is for `currentRecipe` to not be reset (and for `mana` to be reset). Time to look at the code for how that happens.

And I just answered a question of mine, and then had to come back to this paragraph later and rewrite it because although it does only subtract as much mana as it uses the line `this.mana = Math.min(this.mana + mana, manaToGet);` means that it immediately discards any extra mana the moment it arrives. I guess probably because it caused the rhythm to be chaotic and unpleasant to listen to. And in fairness, I prefer that.

And ... here's the relevant code:

`world.spawnEntity(outputItem);
currentRecipe = null;`

It literally resets the recipe the line after spawning the output block. But it doesn't yet clear the inventory, that happens right after. Technically speaking, if an update happened right between those two events, it would set `currentRecipe` again before the inventory was cleared, but that race condition occurring every time would be completely ridiculous, unless something in that code triggered an `updateRecipe()` call. It would be safer to set the recipe to null after clearing the inventory.

But another thing I noticed is that the sound for the rune altar starting up is only played if `manaToGet` changes in an `updateRecipe()` call. Which would require that it be set to zero in one update() tick, and then find the same recipe from before on the next update() tick somehow, and that would actually cause it to be played twice with 50ms delay, which should sound pretty odd. Hrm. Rewatching my vids, it seems I hallucinated that. It only plays the sound of finishing a craft. (Might be because I was watching something on Youtube at the same time, gotta update that in my bug report).

Hmm ... well it seems very clear that somehow `currentRecipe` is not being cleared, or is being restored in the same tick it is cleared before the inventory is cleared. That's the only way it could get stuck in this state.

It highly doubt it could be a client-server desync because there is no way that could cause the mana spreaders to fire. I'm pretty sure that's going to be an event generated server side and sent to the clients, and not something that the client would do any sort of prediction on or simulation of. But I also note that `currentRecipe` is not saved in the NBT data, so if the runic altar is unloaded and reloaded in any way, it should restore itself to a usable state.

Also still have to look into why the dispenser crashes the game and using the wand yourself doesn't.

**Comment by CaptainFord** [comment]
The reason why it crashes with a dispenser seems to simply be that when you right-click the altar yourself the call to onWanded originates from the Network Manager, which presumably catches the Exception, versus it being generated by minecraft's worldTick event. That's my best guess because I don't see a try-catch block anywhere.

Also, I'm wrong about resetting if the altar is unloaded, which is weird. I need to look at how it saves state again.

Wait, hold on, what the actual frick?!?

The line that it's crashing on is when it invokes `recipe.getManaUsage();`, and if `currentRecipe` is not null, that's what it uses. Which means that `currentRecipe` has to be null. But if that were true, the altar would fix itself immediately on the first tick. But that's the only way to get that crash.

Huh?!?

In other words, I now know enough to know that this doesn't make any sense at all. For this to happen, the Runic Altar would need to stop receiving world ticks, I guess. But then it wouldn't be able to update its comparator state.

Also I'm looking at a build of Botania that's a little behind the one being used in Interactions. Hrm.

Well, anyways, that ends my investigation. Without an actual build of Interactions in front of me, I really can't do much more investigation than that. What I'm observing simply doesn't match up with the code I can see.

**Comment by CaptainFord** [comment]
I just realized in the code I'm looking at that `currentRecipe` is actually never set. Not by anything, as far as I can tell. So that theory was busted from the beginning. I just assumed that because it checked it wasn't null that it was an obvious caching behavior, but it turns out no, that was obviously how it worked at one point but there's no actual line that sets the field's value.

So according to the code of Botania 1.12.x final, the behavior we're seeing is outright impossible.

**Comment by ArtpokeBr** [comment]
Is this Issue still relevant in the latest released versions of the Pack? =)

Edit: Ok this seems to possibly related to [this](https://github.com/FTBTeam/FTB-Modpack-Issues/issues/6528) so I'll be closing this issue =)
